src/app/assessment/assessment.component.ts

```diff
@@ -218,6 +219,24 @@ export class AssessmentComponent extends RouterEnter {
     return requiredQuestions;
   }
 
+  getUnansweredCompulsoryQuestion(answers) {
+    const result = [];
+    const missing = [];
+    const required = this.getRequiredQuestions();
```

Comment:
Since we are using `getRequiredQuestions()` only here, we can just move the logic to loop through question from `getRequiredQuestions()` to here instead, so that we don't need to loop twice.